test/test_distribution_generic.py

```diff
@@ -0,0 +1,433 @@
+    # TODO compare sample values on jax backend
+    check_funsor(sample_value, expected_inputs, funsor.Real)
+
+
+@pytest.mark.parametrize("case", TEST_CASES, ids=str)
+@pytest.mark.parametrize("statistic", [
+    "mean",
+    "variance",
+    pytest.param("entropy", marks=[pytest.mark.skipif(get_backend() == "jax", reason="entropy not implemented")])
+])
+def test_generic_stats(case, statistic):
+
+    raw_dist = eval(case.raw_dist)
+
+    dim_to_name, name_to_dim = _default_dim_to_name(raw_dist.batch_shape)
+    with interpretation(lazy):
+        funsor_dist = to_funsor(raw_dist, output=funsor.Real, dim_to_name=dim_to_name)
+
+    with xfail_if_not_implemented(msg="entropy not implemented for some distributions"):
+        actual_stat = getattr(funsor_dist, statistic)()
+
+    expected_stat_raw = getattr(raw_dist, statistic)
+    if statistic == "entropy":
+        expected_stat = to_funsor(expected_stat_raw(), output=funsor.Real, dim_to_name=dim_to_name)
+    else:
+        expected_stat = to_funsor(expected_stat_raw, output=case.expected_value_domain, dim_to_name=dim_to_name)
+
+    check_funsor(actual_stat, expected_stat.inputs, expected_stat.output)
+    if ops.isnan(expected_stat.data).all():
+        pytest.xfail(reason="base stat returns nan")
+    else:
+        assert_close(to_data(actual_stat, name_to_dim), to_data(expected_stat, name_to_dim), rtol=1e-4)
+
+
+@pytest.mark.skipif(not _ENABLE_MC_DIST_TESTS, reason="slow and finicky Monte Carlo tests disabled by default")
+@pytest.mark.parametrize("case", TEST_CASES, ids=str)
+@pytest.mark.parametrize("sample_shape", [(200000,), (400, 400)], ids=str)
+@pytest.mark.parametrize("statistic", [
+    "mean",
+    "variance",
+    pytest.param("entropy", marks=[pytest.mark.skipif(get_backend() == "jax", reason="entropy not implemented")])
+])
+def test_generic_stats_sample(case, statistic, sample_shape):
+
+    raw_dist = eval(case.raw_dist)
+
+    atol = 1e-2
+    with xfail_if_not_implemented(msg="entropy not implemented for some distributions"):
+        actual_stat, expected_stat = _get_stat(raw_dist, sample_shape, statistic)
+
+    check_funsor(actual_stat, expected_stat.inputs, expected_stat.output)
+    if ops.isnan(expected_stat.data).all():
+        pytest.xfail(reason="base stat returns nan")
+    else:
+        assert_close(actual_stat.reduce(ops.add), expected_stat.reduce(ops.add), atol=atol, rtol=None)
+
+
+@pytest.mark.skipif(not _ENABLE_MC_DIST_TESTS, reason="slow and finicky Monte Carlo tests disabled by default")
+@pytest.mark.skipif(get_backend() != "torch", reason="not working yet on jax")
+@pytest.mark.parametrize("case", TEST_CASES, ids=str)
+@pytest.mark.parametrize("sample_shape", [(200000,), (400, 400)], ids=str)
+@pytest.mark.parametrize("statistic", [
+    "mean",
+    "variance",
+    pytest.param("entropy", marks=[pytest.mark.skipif(get_backend() == "jax", reason="entropy not implemented")])
+])
+def test_generic_grads_sample(case, statistic, sample_shape):
```

Comment:
Ditto for this test comparing Monte Carlo estimates of gradients of summary statistics wrt parameters versus gradients of ground-truth values - it's slow and finicky, and I'm not sure failures are very informative.  It's disabled by default, but I am open to removing it entirely.